### PR TITLE
tranjg code review

### PR DESCRIPTION
**Pull request by tranjg**
I've added some comments to the commits I made, so be sure to check that out when you get the chance.

Overall, the app you guys are developing is coming into place, I can see that you guys have a database created, and it looks like you're starting to create functionality to fetch data from the database which is good. I appreciate you guys getting your changes in on time so I could review them. Like I mentioned in one of my comments, make sure that when you're pushing changes to make sure that there aren't any errors with building so that you can have a much smoother developing experience. The changes I made were more changes to the UI as you have the elements rendered, but they were just a bit messy. I also added some boilerplates such as an interface to help you move forward with development.

Here are the commits I did for my group:
https://github.com/jtran549/DynamicDiet/pull/13/commits/8410a4567053bf617f21aeb7cb2a72ed45670278
https://github.com/jtran549/DynamicDiet/pull/13/commits/d6490a13f65cd814200fce8134a0b934d9987f3f
https://github.com/jtran549/DynamicDiet/pull/13/commits/534871f3691afa16b9b2d96ffd40fc7b5bbeb718
https://github.com/jtran549/DynamicDiet/pull/13/commits/7103f35152cc2d55688a5163294b88a31d2b468b

Three technical concepts I learned:
1. Though the Firebase auth you guys are trying to implement isn't finished, I learned a lot about how to use it based on the Firebase documentation and what you guys have implemented. My group is trying to do something similar and thinking about adding Firebase auth as well.
2. I like how you guys have tests involved, it's definitely something my group needs to add. From what you guys did, I took away how to 1. build tests and 2. ideas on what to test for.
3. Based on the code provided, I can assume that you guys are going to have some sort of navigation feature aka navigating between different screens. it helped me to think about how to organize the code in terms of navigation but also gave me ideas on how to go about it.

### CODE REVIEW

**Comment by tranjg** [comment]
https://github.com/Smoofington/FindMyRecyclingV32001/pull/49/commits/58cd11c76a8daccbcfcfacf202d76456b26d1831

Make sure when you're pushing your changes to make sure that the code is functional and that there aren't any errors so that your app can build especially for the other developers.

**Comment by tranjg** [comment]
https://github.com/Smoofington/FindMyRecyclingV32001/pull/49/commits/fe65e6730b2b92d3f2b6a91169066de0008b8da4

I fixed the UI a bit because things looked a bit messy. I placed your button and RecycleSearch into DefaultPreview and that should help you give an idea of how to organize the UI. I'd say when your rendering out elements to have a composable that consists of functions that you want to include for that specific page and those functions should include generalized elements such as a login form or in your guy's case your RecycleSearch function,

**Comment by tranjg** [comment]
https://github.com/Smoofington/FindMyRecyclingV32001/pull/49/commits/105dd9b6c43b11a3f75ce8b8002ae56f1b7e8e92

I saw that you guys were trying to implement Firebase auth and my group is trying to do something similar. I added an interface as a boilerplate. The reasoning for this interface is so that the app enforces those properties. In your guy's case these properties could be implemented into your ViewModel. Essentially, it helps you get an idea of how the foundation of Firebase auth should look.

File: app/src/main/java/com/findmyrecycling/MainActivity.kt [hunk 1]
```diff
@@ -31,7 +30,7 @@ class MainActivity : ComponentActivity() {
                 // A surface container using the 'background' color from the theme
                 Surface(color = MaterialTheme.colors.background,
                 modifier = Modifier.fillMaxWidth()) {
-                    RecycleSearch("Android")
+                    DefaultPreview()
```
Comment (discospiff) [inline app/src/main/java/com/findmyrecycling/MainActivity.kt:33]: Function names should be descriptive... I'm not sure that one of these is more descriptive than the other.

File: app/src/main/java/com/findmyrecycling/MainActivity.kt [hunk 2]
```diff
@@ -75,38 +78,48 @@ fun RecycleSearch(product: String) {
...
+
 fun signIn() {
     val providers = arrayListOf(AuthUI.IdpConfig.EmailBuilder().build())
     val signinIntent = AuthUI.getInstance()
         .createSignInIntentBuilder()
         .setAvailableProviders(providers)
         .build()
 
-    signInLauncher.launch(signinIntent)
+//    signInLauncher.launch(signinIntent)
```
Comment (discospiff) [inline app/src/main/java/com/findmyrecycling/MainActivity.kt:110]: Avoid pushing commented code to Version Control.  Version control has tools that makes this unnecessary.
- You can look at history to see what used to be there.
- You can use branches for experimental features.

File: app/src/main/java/com/findmyrecycling/MainActivity.kt [hunk 3]
```diff
@@ -75,38 +78,48 @@ fun RecycleSearch(product: String) {
...
-private val signInLauncher = registerForActivityResult (
-    FirebaseAuthUIActivityResultContract()
-) {
-        res -> this.signInResult(res)
-}
+//private val signInLauncher = registerForActivityResult (
+//    FirebaseAuthUIActivityResultContract()
+//) {
+//        res -> this.signInResult(res)
+//}
```
Comment (discospiff) [inline app/src/main/java/com/findmyrecycling/MainActivity.kt:117]: Avoid pushing commented code to Version Control.  Version control has tools that makes this unnecessary.
- You can look at history to see what used to be there.
- You can use branches for experimental features.

**Comment by discospiff** [comment]
I'm not entirely clear on how these changes reduce technical debt.  I encourage the reviewer to cite best practice sources in the review, to make a good case for this.  Thus... I'd hold off on merging this branch, until we understand how it decreases technical debt.

**Pull request closed without merging**